DicomRtImportExport/Logic/vtkSlicerDicomRtImportExportModuleLogic.cxx

```diff
@@ -2021,8 +2021,8 @@ std::string vtkSlicerDicomRtImportExportModuleLogic::ExportDicomRTStudy(vtkColle
         vtkSegment* segment = segmentationNode->GetSegmentation()->GetSegment(*segmentIdIt);
 
         // Get binary labelmap representation
-        vtkOrientedImageData* binaryLabelmap = vtkOrientedImageData::SafeDownCast(
-          segment->GetRepresentation(vtkSegmentationConverter::GetSegmentationBinaryLabelmapRepresentationName()) );
+        vtkSmartPointer<vtkOrientedImageData> binaryLabelmap = vtkSmartPointer<vtkOrientedImageData>::Take(
+          segmentationNode->GetBinaryLabelmapRepresentation(segmentID));

```

Comment:
Is there any documentation in the GetBinaryLabelmapRepresentation and similar functions that now the returned object needs to be deleted?

Reply:
I added this comment to the header file documentation:
https://github.com/Slicer/Slicer/blob/c9203a9f13fdd727965dd7ec6cbb7623863aaa76/Libs/MRML/Core/vtkMRMLSegmentationNode.h#L210

Reply:
The risk of misuse is so high that documentation is not enough. Factory methods must start with a very clear name (usually "Create") but some developers even missed those, so now when we need to assign ownership we always use `GetSomething(vtkSomeObject *objCreatedByCaller)` pattern.

If changing the method signature would cause too much inconvenience for the users then these labelmap representations could be cached in the segmentation node or object (similarly to how returned vectors are cached in classes to allow simple _spacing=img->GetSpacing()_ syntax)

Reply:
I usually use this kind of signature in such cases:

```
vtkNew<vtkOrientedImageData> binaryLabelmap;
segmentationNode->GetBinaryLabelmapRepresentation(segmentID, binaryLabelmap);
```

Reply:
Ok, I'll change the function signature in the segmentation node.

Do you want me to change the GetClosedSurfaceRepresentaion signature for consistency?
It is not currently a factory method.